src/login/Login.cpp

```diff
@@ -267,3 +354,1805 @@ void ProfileRecord::FormatTo(char* buffer, size_t length) const
 
 	*result.out = 0;
 }
+
+class WithDb
+{
+public:
+	explicit WithDb(const int flags)
+	{
+		if (sqlite3_open_v2(s_dbPath.c_str(), &m_db, flags, nullptr) != SQLITE_OK)
+		{
+			SPDLOG_ERROR("AutoLogin Error failed to open database %s: %s", s_dbPath, sqlite3_errmsg(m_db));
+			sqlite3_close(m_db);
+			m_db = nullptr;
+		}
+		else
+		{
+			// pragmas don't need return values, just execute them
+			sqlite3_exec(m_db, "PRAGMA journal_mode = WAL", nullptr, nullptr, nullptr);
+			sqlite3_exec(m_db, "PRAGMA synchronous = normal", nullptr, nullptr, nullptr);
+			sqlite3_exec(m_db, "PRAGMA temp_store = memory", nullptr, nullptr, nullptr);
+			sqlite3_exec(m_db, "PRAGMA mmap_size = 100000000", nullptr, nullptr, nullptr);
+			sqlite3_exec(m_db, "PRAGMA foreign_keys = ON", nullptr, nullptr, nullptr);
+		}
+	}
+
+	~WithDb()
+	{
+		if (m_db != nullptr) sqlite3_close(m_db);
+	}
+
+	[[nodiscard]] sqlite3* GetDB() const { return m_db; }
+
+	template <typename T>
+	static std::function<T(const std::string&, const std::function<T(sqlite3_stmt*, sqlite3*)>&)> Query(int flags)
```

Comment:
Why are we currying flags? Why not just take the query and action as parameters?

Reply:
at one point, I was storing this function per flags, but when the interface changed I never changed it